src/components/Basket/Basket.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { getBasket } from '../../store/index';
+import { GoodsList } from './GoodsList';
+
+export const Basket: React.FC = () => {
+  const goodsList = useSelector(getBasket);
+  const totalQuantity = goodsList.reduce((acc, value) => acc + value.quantity, 0);
+  const totalPrice = goodsList.reduce(
+    (acc, value) => acc + value.quantity * value.goodItem.price
+    * (1 - value.goodItem.discount / 100), 0,
+  );
```

Comment:
читаемость под вопросом
```suggestion
  const totalPrice = goodsList.reduce((acc, value) => {
    const value = acc + value.quantity * value.goodItem.price;
    
    return value * (1 - value.goodItem.discount / 100);
  }, 0);
```

задача: разделить вычисления, value можно заменить по контексту